src/Integration.Vsix/CFamily/VcxProject/VcxRequestFactory.cs

```diff
@@ -44,33 +44,39 @@ internal class VcxRequestFactory : IRequestFactory
     {
         private readonly ICFamilyRulesConfigProvider cFamilyRulesConfigProvider;
         private readonly IThreadHandling threadHandling;
-        private readonly IFileConfigProvider fileConfigProvider;
+        private readonly Lazy<IFileConfigProvider> fileConfigProvider;
         private readonly ILogger logger;
-        private readonly DTE2 dte;
+        private readonly Lazy<DTE2> dte;
 
         [ImportingConstructor]
         public VcxRequestFactory([Import(typeof(VsShell.SVsServiceProvider))] IServiceProvider serviceProvider,
             ICFamilyRulesConfigProvider cFamilyRulesConfigProvider,
-            ILogger logger)
+            ILogger logger,
+            IThreadHandling threadHandling)
             : this(serviceProvider,
                 cFamilyRulesConfigProvider,
-                new FileConfigProvider(logger),
+                new Lazy<IFileConfigProvider>(() => new FileConfigProvider(logger)),
                 logger,
-                ThreadHandling.Instance)
```

Comment:
Moved this up because it is mef importable, this way its more consistent to all the other classes